sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -733,6 +763,10 @@ static int sof_audio_probe(struct platform_device *pdev)
 	dmi_check_system(sof_rt5682_quirk_table);
 
 	mach = pdev->dev.platform_data;
+	dmi_check_system(sof_rt5682_tplg_table);
+
+	if(sof_override_tplg_name)
+		mach->sof_tplg_filename=sof_override_tplg_name;
```

Comment:
wouldn't it be possible to do:
```
struct rt5682_tplg_driver {
	const char *tplg;
	bool override;
};
struct rt5682_tplg_driver ddata = {
	.tplg = "sof-tgl-max98373-rt5682-ampI2S2.tplg",
};
const struct dmi_system_id sof_rt5682_tplg_table[] = {
	{
		.callback = sof_rt5682_tplg_cb,
		.matches = {
			/* will replace with real product name later*/
			DMI_MATCH(DMI_PRODUCT_NAME, "Google Unique Product Name"),
		},
		.driver_data = &ddata,
	},
	{}
};
dmi_check_system(&ddata);
if (ddata.override)
	mach->sof_tplg_filename = ddata.tplg;
```
and you'd need to modify the callback too of course.

Reply:
I checked this, it is not possible because to dmic_check_system we can not pass ddata structure.

we have to pass dmi table list:
dmi_check_system(const struct dmi_system_id *list)